Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs

```diff
@@ -36,125 +41,348 @@ public class CLIProcessor
+            if (runOptions.Environment == null)
+            {
+                sb.Append(" auto select environment (assuming only one exist)");
+            }
+            else
+            {
+                sb.Append(" select environment '" + runOptions.Environment + "'");
+            }
+                            
+            sb.Append(" and execute run set '").Append(runOptions.Runset).Append("'");
+            sb.Append(Environment.NewLine);
+            sb.Append("Long form : ").Append("'" + exe + arguments + "'").Append(Environment.NewLine);
+            sb.Append("Short form: ").Append("'" + exe + "run");
+            sb.Append(" -").Append(CLIOptionClassHelper.GetAttrShorName<RunOptions>(nameof(RunOptions.Solution))).Append(" ").Append(WrapTextIfNeeded(runOptions.Solution));
+            if (env != null)
+            {
+                sb.Append(" -").Append(CLIOptionClassHelper.GetAttrShorName<RunOptions>(nameof(RunOptions.Environment))).Append(" ").Append(WrapTextIfNeeded(runOptions.Environment));
             }
+
+            if (runset != null)
+            {
+                sb.Append(" -").Append(CLIOptionClassHelper.GetAttrShorName<RunOptions>(nameof(RunOptions.Runset))).Append(" ").Append(WrapTextIfNeeded(runOptions.Runset));
+            }
+
+            sb.Append("'");
+            sb.Append(Environment.NewLine);
+            sb.Append(Environment.NewLine);
+            return sb.ToString();
         }
 
-        private void PerformLoadAndExecution(string configurations)
+        string WrapTextIfNeeded(string val)
         {
-            Reporter.ToLog(eLogLevel.DEBUG, "Loading Configurations...");
-            mCLIHandler.LoadContent(configurations, mCLIHelper, WorkSpace.Instance.RunsetExecutor);
+            if (val.IndexOf(' ') > 0)
+            {
+                return "\"" + val + "\"";
+            }
+            else
+            {
+                return val;
+            }
+        }
+
+        private int HandleSCMOptions(SCMOptions sCmOptions)
+        {
+            SetVerboseLevel(sCmOptions.VerboseLevel);
+            Reporter.ToLog(eLogLevel.INFO, "Running SCM options");
+
+            mCLIHandler = new CLISCM();
+            mCLIHelper.SourceControlURL = sCmOptions.URL;
+            
+            // do the rest !!!
+            
+            ExecuteRunSet();
+
+            return Environment.ExitCode;
+        }
+
+        private int HandleScriptOptions(ScriptOptions scriptOptions)
+        {
+            SetVerboseLevel(scriptOptions.VerboseLevel);
+
+            Reporter.ToLog(eLogLevel.DEBUG, string.Format("Running with ScriptFile= '{0}'", scriptOptions.FileName));
+            mCLIHandler = new CLIScriptFile();
+            mCLIHandler.LoadContent(ReadFile(scriptOptions.FileName), mCLIHelper, WorkSpace.Instance.RunsetExecutor);                        
+            ExecuteRunSet();
+            
+            return Environment.ExitCode;
+        }
+
+        
```

Comment:
as disscused, code for handling Config/Dynamic/Script/Run is mostly the same, i think we should group it into single function with Switch case for creating the Object (and also for parsing the Args in case of Run)- rest is same for all- will be easier to maintain

Reply:
Combined to one method please review